File: leetcode/125/125.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        lower_s = s.lower()
        lower_s = lower_s.strip()
        reversed_lower_s = "".join(filter(str.isalnum, lower_s))
        if reversed_lower_s[::-1] == reversed_lower_s:
            return True
        return False
        
sol = Solution()
print(sol.isPalindrome("A man, a plan, a canal: Panama"))

File: leetcode/125/125.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
    var filtered = s.toLowerCase().replace(/[^a-z0-9]/g, '')
    let left = 0
    let right = filtered.length -1
    while(left < right){
        if(filtered[left]!==filtered[right]){
            return false
        }
        left++
        right--
    }
    return true
};

console.log(isPalindrome("A man, a plan, a canal: Panama")); // true
console.log(isPalindrome("race a car")); // false
